src/js/actions/MesosActions.js

```diff
@@ -0,0 +1,48 @@
+var AppDispatcher = require("../AppDispatcher");
+var JSONPUtil = require("../helpers/JSONPUtil");
+var MesosEvents = require("../events/MesosEvents");
+
+var MesosActions = {
+  requestState: function (id, url) {
+    JSONPUtil.request(`${url}/state`).then(
```

Comment:
Ah ok, then you need to detect the mesos version somehow, to be compatible to all these versions.